### PR TITLE
Show ibus methods

### LINKED ISSUE #80: No keyboard/language indicator.
There is no keyboard or language indicator on the top panel at the right side corner.
I use US English and Greek keyboard and I have a serious problem in changing languages.
Thank you

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/94405770-no-keyboard-language-indicator?utm_campaign=plugin&utm_content=tracker%2F60236121&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F60236121&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #80 

Relies on https://github.com/elementary/switchboard-plug-keyboard/pull/341 which must be merged first.
Include ibus input methods in the wingpanel indicator and switches between them and xkb layouts using the `<Alt><Shift>` shortcut or choosing from the popover.

NOTE: Ibus methods only appear in the wingpanel indicator when the ibus daemon is running.  On my system this does not happen automatically without a custom startup command.  Should the indicator start up the daemon??

### CODE REVIEW

**Review by ryonakano — COMMENTED** [review]
Nice improvements! :tada:

File: src/LayoutsManager.vala [hunk 1]
```diff
@@ -41,48 +53,102 @@ public class Keyboard.Widgets.LayoutManager : Gtk.ScrolledWindow {
...
-                    string language;
-                    string? variant = null;
-                    if ("+" in source) {
-                        var layouts = source.split ("+", 2);
-                        language = layouts[0];
-                        variant = layouts[1];
-                    } else {
-                        language = source;
-                    }
+            string name = "English";
```
[the remark this answers is not in the document]
Comment (jeremypw) [reply src/LayoutsManager.vala]: I have reworked handling of fallbacks now making this translation unnecessary.

File: src/LayoutsManager.vala [hunk 2]
```diff
@@ -134,53 +231,19 @@ public class Keyboard.Widgets.LayoutManager : Gtk.ScrolledWindow {
         Xml.Node* node = res->nodesetval->item (0);
         if (node != null) {
             name = dgettext ("xkeyboard-config", node->get_content ());
-        }
+        }//~     private LayoutButton? get_current_layout_button () {
```
[the remark this answers is not in the document]
Comment (jeremypw) [reply src/LayoutsManager.vala]: Oops - now gone.

**Comment by ryonakano** [comment]
I'm using a Japanese keyboard and expect the keyboard layout should always be set to Japanese even if I use Japanese IME (Mozc), and this is the current behavior with the latest master. However, with this branch the keyboard layout seems to be specified to English (US) when I set the current engine to Mozc.

**Comment by jeremypw** [comment]
@ryonakano So we need to show/control both which physical keyboard layout is in use *and* which input method is in use?  At the moment the ibus settings should control which keyboard layout is used when a particular input method is selected.  It might be useful to be able to launch the ibus settings app from the indicator (and maybe the daemon as well if it is not running).

**Comment by jeremypw** [comment]
@ryonakano Do have "Use system keyboard layout" checked in ibus preferences?

**Comment by jeremypw** [comment]
@ryonakano This may help with your issue: https://askubuntu.com/questions/1029522/japanese-mozc-wrong-keyboard-layout

I am not sure why this PR should have changed the way the ibus-mozc engine is working.

**Comment by jeremypw** [comment]
Closing at it relies on closed PR.

**Pull request closed without merging**